cmd/formula-analytics.rb

```diff
@@ -355,59 +355,75 @@ def influx_analytics(args)
 
     odie "No InfluxDB credentials found in HOMEBREW_INFLUXDB_TOKEN!" unless ENV["HOMEBREW_INFLUXDB_TOKEN"]
 
-    # TODO: attempt to strip out the date and ideally index columns
-    # TODO: attempt to sort on server-side
-    # TODO: handle other dimensions
-    query = <<~EOS
-      from(bucket: "analytics")
-        |> range(start: -31d, stop: now())
-        |> filter(fn: (r) => r["_measurement"] == "formula_install")
-        |> group(columns: ["package_and_options"])
-        |> sum()
-    EOS
-    result = influxdb_client.create_query_api.query_raw(query: query)
-    # remove datatype, group, empty, column name lines
-    lines = result.lines.drop(4)
-    json = {
-      category:    :formula_install,
-      total_items: 0,
-      start_date:  nil,
-      end_date:    nil,
-      total_count: 0,
-      items:       [],
-    }
-
-    lines.each do |line|
-      _, _, _index, start_date, end_date, count, name = line.split(",")
-      next if name.blank?
-
-      count = count.to_i
-
-      json[:total_items] += 1
-      json[:start_date] ||= DateTime.parse(start_date)
-      json[:end_date] ||= DateTime.parse(end_date)
-      json[:total_count] += count
-
-      json[:items] << {
-        number:  nil,
-        # TODO: handle other dimensions
-        formula: name.chomp,
-        count:   count,
-      }
+    days_ago = (args.days_ago || 30).to_i
+    if days_ago > 365
+      opoo "Analytics only retained for 1 year `--days-ago` set to maximum value."
+      days_ago = 365
     end
 
-    json[:items].sort_by! do |item|
-      -item[:count]
-    end
+    categories = []
+    categories << :formula_install if args.install?
+    categories << :cask_install if args.cask_install?
+    categories << :build_error if args.build_error?
+    categories << :os_versions if args.os_version?
 
-    json[:items].each_with_index do |item, index|
-      item[:number] = index + 1
+    field = "package"
+    tag = "pkg"
 
-      percent = (item[:count].to_f / json[:total_count]) * 100
-      item[:percent] = format_percent(percent)
-    end
+    # TODO: handle on_request and OS version dimensions
+    categories.each do |category|
+      field = "os_name_and_version" if category == :os_versions
+      tag = "os" if category == :os_versions
+      query = <<~EOS
+        from(bucket: "analytics_downsampled")
```

Comment:
Yeah, we have a task doing hourly calculations and it writes to this new bucket. From there it's trivial to sum them up as you can see here.

Reply:
Would be useful to note down how the tasks were set up, even if that's only internal somewhere.

It's just in case we need to set it up again in the future (unlikely but possible).

Reply:
Yeah, not sure where to do that. I'll think of something for a next PR though.

Reply:
Agreed. I reckon once it's all working: let's try to get documented.



> Yeah, we have a task doing hourly calculations and it writes to this new bucket. From there it's trivial to sum them up as you can see here.

These could be daily or something if it's as easy and somehow saves us resources/money/anything?

Reply:
They could be, but it doesn't affect anything and hours give us some more granularity, faster feedback for me and quicker runtime overall.

Reply:
Yup, sounds like an obvious decision then, resolving!